pkg/engine/executor/expressions.go

```diff
@@ -86,77 +100,90 @@ type ColumnVector interface {
 	ToArray() arrow.Array
 	// Value returns the value at the specified index position in the column vector.
 	Value(i int) any
-	// Type returns the Arrow data type of the column vector.
-	Type() arrow.DataType
+	// ArrowType returns the Arrow data type of the column vector.
+	ArrowType() arrow.DataType
```

Comment:
nit: What if we moved `ArrowType()` from a method on `ColumnVector` to one on `datatype.DataType`?

This would decouple the transport (arrow) from the internal vector by making it derivable by our dataypes instead.

The behavior is equivalent, but it feels a bit nicer (IMO) to decouple these, especially if we'll need to add an arrow alternative at some point. WDYT? This is a small preference and should in no way block.

Reply:
Good idea!